### PR TITLE
Remove Mailu PostgreSQL

### LINKED ISSUE #1838: [Setup utility] Cannot generate files when database flavors have been switched
## Before you open your issue
- [x] Check if no issue or pull-request for this already exists.
- [x] Check [documentation](https://mailu.io/master/) and [FAQ](https://mailu.io/master/faq.html). (Tip, use the search function on the documentation page)
- [x] You understand `Mailu` is made by volunteers in their **free time** — be conscise, civil and accept that delays can occur.
- [x] The title of the issue should be short and simple. It should contain specific terms related to the actual issue. Be specific while writing the title.

### Environment
 - [x] docker-compose
 - [ ] kubernetes
 - [x] docker swarm

### Versions
All versions

## Description
In the setup utility when you select a different database flavor and then switch back to SQLite, then you can not generate the files. The reason is that it still expects the fields to be filled of the alternative database flavor you previously selected. 
When you select an alternative database flavor, the fields (e.g. hostname, database name) are mandatory. These fields are still mandatory when you switch back to SQLlite as database flavor.

As a workaround you can fill in mandatory fields and then switch back to SQLite again. Or you could refresh the page and do not switch from SQLite. 

The problem is in https://github.com/Mailu/Mailu/blob/master/setup/templates/steps/database.html . 
I didn't check yet how to resolve this.

## Replication Steps
1) Go to https://setup.mailu.io/master/ and click next.
2). Enter a value for main domain server and public hostname.
3). Select roundcube as webmail.
4). At the bottom switch to postgresql or mysql as database
5). Switch back to SQLite as database.
6). Click Setup mailu. Note that the button does not work.

## Expected behaviour
The Setup Mailu button works after following about steps.

### PR DESCRIPTION
**Pull request by Diman0**
## What type of PR?

enhancement

## What does this PR do?
- Removes Mailu PostgreSQ
- Makes roundcube database configurable via setup
- Fixes bug #1838 

As already announced in the release of Mailu 1.8, we will remove Mailu PostgreSQL in 1.9. In PR #2064 we created the necessary documentation to assist migrating to the official PostgreSQL image. This PR completely removes Mailu PostgreSQL. As a bonus I fixed bug #1838. People who were using PostgreSQL with Roundcube are in the situation that Roundcube must keep using SQLite. Roundcube has no migration or export/import functionality to move to a different database flavour (what we do have :). Therefore I have made the roundcube database selection configurable in setup. 

### Related issue(s)
- closes #1838

## Prerequisites
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [n/a] In case of feature or enhancement: documentation updated accordingly - will happen in Mailu 1.9 release notes.
- [x] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

### CODE REVIEW

**Comment by Diman0** [comment]
bors try

File: setup/static/render.js [hunk 1]
```diff
@@ -35,80 +35,57 @@ $(document).ready(function() {
...
-		} else if (this.value == 'mysql') {
-			$("#postgres_db").hide();
+			$("#db_user").prop('required',false);
+			$("#db_pw").prop('required',false);
+			$("#db_url").prop('required',false);
+			$("#db_name").prop('required',false);
+			$("#roundcube_db_user").prop('required',false);
+			$("#roundcube_db_pw").prop('required',false);
+			$("#roundcube_db_url").prop('required',false);
+			$("#roundcube_db_name").prop('required',false);
```
Comment (ghostwheel42) [inline setup/static/render.js:65]: why this? I think this should be removed.

**Review by ghostwheel42 — DISMISSED** [review]
I've added some comments just by reading the code. I didn't test-run it yet.

File: setup/static/render.js (hunk 1 above)
Comment (Diman0) [reply setup/static/render.js:65]: This is the fix for #1838. If these **required** flags are not set to false, you cannot click on Submit after switching back to SQLite.
The reason for this is that the hidden database form fields still expect all fields to be filled in  (they are still set to required,true). Setting the required property to false allows you to submit the form after switching the db flavour back to SQLite.

File: setup/static/render.js [hunk 2]
```diff
@@ -35,80 +35,57 @@ $(document).ready(function() {
...
-	}
-	if ($('#external_psql').prop('checked')) {
+		$("#external_db").hide();
+	} else if ($("#database").val() != 'sqlite') {
 		$("#external_db").show();
 	}
+
+	$("#webmail").click(function() {
+		if (this.value == 'roundcube') {
+			$("#db_flavor_rc_sel").show();
```
[the remark this answers is not in the document]
Comment (Diman0) [reply setup/static/render.js:45]: That is actually set here:
https://github.com/Mailu/Mailu/blob/9eba2feae2053d7a8c64a8b569edfe248c4cd58c/setup/static/render.js#L75
The default  database flavour is SQLite for Roundcube which does **not** require these props set to **required** yet. Doing what you suggest results in not being able to submit the form if you use SQLite for roundcube (due to having invisible fields set to required).

File: setup/static/render.js [hunk 3]
```diff
@@ -35,80 +35,57 @@ $(document).ready(function() {
...
-		} else {
-			$("#external_db").hide();
+
+	$("#database_rc").click(function() {
+		if (this.value == 'sqlite'){
+			$("#roundcube_external_db").hide();
+			$("#roundcube_db_user").prop('required',false);
+			$("#roundcube_db_pw").prop('required',false);
+			$("#roundcube_db_url").prop('required',false);
+			$("#roundcube_db_name").prop('required',false);
```
[the remark this answers is not in the document]
Comment (Diman0) [reply setup/static/render.js:81]: That is good to know. I will apply this.

File: setup/static/render.js [hunk 4]
```diff
@@ -35,80 +35,57 @@ $(document).ready(function() {
...
 	if ($("#database").val() == 'sqlite') {
-		$("#postgres_db").hide();
-	} else if ($("#database").val() == 'postgresql') {
-		$("#postgres_db").show();
-	} else if ($("#database").val() == 'mysql') {
-		$("#external_db").show();
-	}
-	if ($('#external_psql').prop('checked')) {
+		$("#external_db").hide();
+	} else if ($("#database").val() != 'sqlite') {
```
[the remark this answers is not in the document]
Comment (Diman0) [reply setup/static/render.js:39]: True. I will apply this

File: setup/static/render.js [hunk 5]
```diff
@@ -35,80 +35,57 @@ $(document).ready(function() {
...
-			$("#postgres_db").hide();
+			$("#db_user").prop('required',false);
+			$("#db_pw").prop('required',false);
+			$("#db_url").prop('required',false);
+			$("#db_name").prop('required',false);
+			$("#roundcube_db_user").prop('required',false);
+			$("#roundcube_db_pw").prop('required',false);
+			$("#roundcube_db_url").prop('required',false);
+			$("#roundcube_db_name").prop('required',false);
+		} else if (this.value != 'sqlite') {
```
[the remark this answers is not in the document]
Comment (Diman0) [reply setup/static/render.js:66]: True. I will apply this

File: setup/static/render.js [hunk 6]
```diff
@@ -35,80 +35,57 @@ $(document).ready(function() {
...
-			$("#external_db").hide();
-			$("#db_user").val("");
-			$("#db_pw").val("");
-			$("#db_url").val("");
-			$("#db_name").val("");
-			$("#db_user").prop('required',false);
-			$("#db_pw").prop('required',false);
-			$("#db_url").prop('required',false);
-			$("#db_name").prop('required',false);
+		if (this.value != 'sqlite' && $("#webmail").val() == 'roundcube') {
```
[the remark this answers is not in the document]
Comment (Diman0) [reply setup/static/render.js:83]: True. I will apply this

**Comment by Diman0** [comment]
Testing this is really annoying. The only way to convince Firefox **NOT** to use the cache is to create a private tab. And when I want to test again, I must close all private tabs and open private tabs again. This is the only way for me to prevent annoying caching issues :cry: 

I applied your remarks. It looks cleaner now. All tests in setup still work fine for me.

bors try

**Comment by nextgens** [comment]
> Testing this is really annoying. The only way to convince Firefox **NOT** to use the cache is to create a private tab. And when I want to test again, I must close all private tabs and open private tabs again. This is the only way for me to prevent annoying caching issues cry

Actually, all you need is the dev tools: press F12 then Network -> disable cache

**Comment by Diman0** [comment]
resolved merge conflicts. 

bors try

**Review by nextgens — APPROVED** [review]

**Pull request merged**